README.md

```diff
@@ -17,3 +17,329 @@ Implement a prototype job worker service that provides an API to run arbitrary L
+    repeated ProcessInfo processTable = 1;
+}
+
+type QueryRunningProcessesResponse {
+     ProcessInfo[] processTable = 1;
+}
+type QueryRequest {
+    variableField queryCommand {
+        QueryPidRequest queryPid 
+        QueryRunningProcessesRequest queryRunningProcesses 
+        QueryListOfLogsResponse responseFromListOfLogs
+    }
+}
+
+type QueryResponse {
+variableField responseFromQuery {
+        QueryPidResponse responseFromQueryPid
+        QueryRunningProcessesResponse responseFromRunningProcesses
+        QueryListOfLogsResponse responseFromListOfLogs 
+    }
+}
+
+```
+
+
+### Implementation Overview 
+
+* Using GRPC so the above methods and types with be generated via protocol buffers (will be using libprotoc 3.11.4)
+
+####  DataStore
+
+* we can use sync.Map in Go to implement a set of structs to store process info
+
+
+#### Server 
+
+* data store for running pid, starting timestamp, process names
+    
+    * can be implemented with a concurrent map
+
+* function to execute commands 
+
+    * os/exec and syscall packages can take care of this
+
+* logfile system management
+
+    * can use log, writer and reader packages
+
+    * must manage concurrency issues with read and write
+    
+    * will be similar to how linux implements it using the /proc path, but simpler:
+        ```
+        $ ls -al  /proc | head -n 10
+        total 4
+        dr-xr-xr-x 312 root             root                           0 Aug 24 11:41 .
+        drwxr-xr-x  23 root             root                        4096 Aug 17 16:01 ..
+        dr-xr-xr-x   9 root             root                           0 Aug 24 11:41 1
+        dr-xr-xr-x   9 root             root                           0 Aug 26 16:40 10
+        dr-xr-xr-x   9 avwong13         avwong13                       0 Aug 26 16:40 1003
+        dr-xr-xr-x   9 root             root                           0 Aug 26 17:08 1004
+        dr-xr-xr-x   9 avwong13         avwong13                       0 Aug 25 23:08 10202
+        dr-xr-xr-x   9 avwong13         avwong13                       0 Aug 25 23:09 10256
+        dr-xr-xr-x   9 avwong13         avwong13                       0 Aug 24 13:25 10485
+        ```
+
+* The server will only store running pids in memory and have the pass logs available in their respective \<Pid\>_\<startingTimeStamp\> folder
+
+* concurrent go routines to handle cmd executions starting and finishing 
+
+* killing of pid can be implemented can be done through sigkill via syscall
+
+* Execute Api will be a bidirection streaming api, so the server can continously send back responses, while client can send new requests
+
+* Data for query for all processes ever executed by linux worker server should be generated by iterating through file system and extracting data from
+logs and filepath using path packages
+
+* Implement simple logging for server, as in the server stdout and stderr will output into a log in the server's filesystem
```

Comment:
don't worry about the logs of the server itself, they can go to stdout/stderr
if needed, something like systemd can redirect those to a file.